### PR TITLE
Feature 59 manually sync

### LINKED ISSUE #59: manually sync data
As a user, I wish to have the ability to sync the data manually

### PR DESCRIPTION
**Pull request by Chi-Chu319**
resolves #59

### CODE REVIEW

File: src/localization/fi.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
     "title": "Aikapankki",
     "logo": "Sivun logo",
     "managementLink": "Hallinta",
-    "logout": "Kirjaudu ulos"
+    "logout": "Kirjaudu ulos",
+    "syncData": "Synkronoida",
+    "syncDataLoading": "Tietojen synkronointi, ole kärsivällinen."
```
[the remark this answers is not in the document]
Comment (EVP1989) [reply src/localization/fi.json:8]: "Synkronoi" as a verb and "odota hetki" = wait a moment, in my opinion, works better on finnish locale.

File: src/generated/client/apis/TimebankApi.ts [hunk 2]
```diff
@@ -129,11 +129,35 @@ export class TimebankApi extends runtime.BaseAPI {
...
+
+        const response = await this.request({
+            path: `/timebank/sync`,
+            method: 'POST',
+            headers: headerParameters,
+            query: queryParameters,
+        });
+
+        return new runtime.TextApiResponse(response) as any;
+    }
```
Comment (EVP1989) [inline src/generated/client/apis/TimebankApi.ts:152]: Why any type?

**Review by EVP1989 — CHANGES_REQUESTED** [review]

File: src/components/layouts/app-layout.tsx [hunk 3]
```diff
@@ -28,12 +32,64 @@ interface Props {
...
+      await Api.getTimeBankApi().timebankControllerSyncWorkTime();
+    } catch (error) {
+      console.log("error", error);
+      context.setError(strings.errorHandling.syncTimeDataFailed, error);
+    }
+
+    if (person) {
+      const personCloned = { ...person } as PersonDto;
+      dispatch(setPerson(undefined));
+      dispatch(setPerson(personCloned));
```
[the remark this answers is not in the document]
Comment (Chi-Chu319) [reply src/components/layouts/app-layout.tsx:57]: To refresh the selected person, the drawer and editor only rerender and update state when you reselect the person (they are configured to listen to the changes for the person state)

File: src/generated/client/apis/TimebankApi.ts (hunk 2 above)
Comment (Chi-Chu319) [reply src/generated/client/apis/TimebankApi.ts:152]: generate by openapi tool, but I think the yaml is outdated so its regenerated

**Review by EVP1989 — COMMENTED** [review]
Looking good

**Review by jarinykanen — CHANGES_REQUESTED** [review]

**Pull request merged**